compose/config/config.py

```diff
@@ -199,14 +238,52 @@ def load(config_details):
 
     Return a fully interpolated, extended and validated configuration.
     """
+    version = get_config_version(config_details)
+    if version not in COMPOSEFILE_VERSIONS:
+        raise ConfigurationError('Invalid config version provided: {0}'.format(version))
```

Comment:
Cool, if for some reason this needs to happen after processing files , the block below that iterates over the config_details could be moved into a function and called from both branches.